libs/sim/simulator.rs

```diff
@@ -166,23 +167,42 @@ impl<'a> Simulator {
 
     // if self.record_network_data is set, returns amt of data used per
     // tick
-    pub fn tick(&mut self, tick: u64) -> usize {
+    pub fn tick(&mut self, tick: u64) {
         log::info!("################# TICK {0} START #################", tick);
         let mut rpc_buffer = vec![];
-        let mut data_over_network_size = 0;
         // tick all elements to generate RPCs
         // this is the send phase. collect all the RPCs
         for (_elem_name, element_obj) in self.elements.iter_mut() {
             let rpcs = element_obj.tick(tick);
             for rpc in &rpcs {
-                if self.record_network_data {
-                    data_over_network_size += rpc.len();
-                }
                 rpc_buffer.push(rpc.clone());
             }
             log::info!("{:45}", element_obj);
             log::info!("\toutputs {:?}", rpcs);
         }
+
+        // collect network data if necessary
+        if self.record_network_data {
+            let mut data_over_network_size = 0;
+            for rpc in &rpc_buffer {
+                data_over_network_size += rpc.len();
+            }
+            match Writer::from_path("network_results.csv") {
+                Ok(mut wtr) => {
+                    match wtr.write_record(&[tick.to_string(), data_over_network_size.to_string()])
+                    {
+                        Ok(_) => {}
+                        Err(e) => {
+                            log::error!("Could not write record, {0}", e);
+                        }
+                    }
+                }
+                Err(e) => {
+                    log::error!("Could not open csv writer, {0}", e);
+                }
+            }
+        }
```

Comment:
I would make this a function and then this should be okay.

Reply:
Sounds good.  I also edited it so that you specify the file name in the arguments.